src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.cpp

```diff
@@ -0,0 +1,228 @@
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/ConstantExpressions.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Math.hpp" // IWYU pragma: keep
+
+namespace {
+template <typename DataType>
+Scalar<DataType> compute_piecewise(
+    const tnsr::I<DataType, 3, Frame::Inertial>& x, const double& inner_radius,
+    const double& outer_radius, const double& inner_value,
+    const double& outer_value) noexcept {
+  const DataType cylindrical_radius =
+      sqrt(square(get<0>(x)) + square(get<1>(x)));
+  auto piecewise_scalar = make_with_value<Scalar<DataType>>(x, 0.0);
+
+  // piecewise_scalar should equal inner_value for r < inner_radius,
+  // should equal outer_value for r > outer_radius, and should transition
+  // in between. Here, use step_function() to compute the result
+  // in each region separately.
+  get(piecewise_scalar) +=
+      step_function(inner_radius - cylindrical_radius) * inner_value;
+  get(piecewise_scalar) +=
+      step_function(cylindrical_radius - outer_radius) * outer_value;
+  get(piecewise_scalar) +=
+      // Blaze's step_function() at 0.0 is 1.0. So to get a mask that
+      // is the inverse of the masks used above for inner_radius and
+      // outer_radius without double-counting points on the boundary,
+      // use 1.0 - stepfunction(x) instead of stepfunction(-x) here
+      (1.0 - step_function(inner_radius - cylindrical_radius)) *
+      (1.0 - step_function(cylindrical_radius - outer_radius)) *
+      exp(((-1.0 * cylindrical_radius + inner_radius) * log(outer_value) +
+           (cylindrical_radius - outer_radius) * log(inner_value)) /
+          (inner_radius - outer_radius));
+  return piecewise_scalar;
+}
+}  // namespace
+
+/// \cond
+namespace grmhd {
+namespace AnalyticData {
+
+CylindricalBlastWave::CylindricalBlastWave(
+    const InnerRadius::type inner_radius, const OuterRadius::type outer_radius,
+    const InnerDensity::type inner_density,
+    const OuterDensity::type outer_density,
+    const InnerPressure::type inner_pressure,
+    const OuterPressure::type outer_pressure,
+    const MagneticField::type magnetic_field,
+    const AdiabaticIndex::type adiabatic_index, const OptionContext& context)
+    : inner_radius_(inner_radius),
+      outer_radius_(outer_radius),
+      inner_density_(inner_density),
+      outer_density_(outer_density),
+      inner_pressure_(inner_pressure),
+      outer_pressure_(outer_pressure),
+      magnetic_field_(magnetic_field),
+      adiabatic_index_(adiabatic_index),
+      equation_of_state_(adiabatic_index) {
+  if (inner_radius >= outer_radius) {
+    PARSE_ERROR(context,
+                "CylindricalBlastWave expects InnerRadius < OuterRadius");
```

Comment:
Is it possible to include the values in the error message?

Reply:
Sure, I would just have to set up a string stream or something, and didn't want to unless I had to, because it's a bit of a pain, and these options are in the input file, so easy to find and change. But I'm happy to add this.

Reply:
you can  stream inside the PARSE_ERROR using `<<`